### PR TITLE
New blog from oVirt infra on 'build-on-demand' feature

### PR DESCRIPTION
**Pull request by eedri**
oVirt infra team added a new feature for
	developers to build artifacts on CI,
	on-demand, using just a comment in the patch.

Signed-off-by: Eyal Edri <eedri@redhat.com>

Fixes issue # (delete if not relevant)

Changes proposed in this pull request:

-

-

-

I confirm that this pull request was submitted according to the [contribution guidelines](/CONTRIBUTING.md): (please @mention yourself to sign)

This pull request needs review by: (please @mention the reviewer if relevant)

### CODE REVIEW

File: source/blog/2016-12-20-ci-please-build.html.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+---
+title: CI Please Build&mdash;How to build your oVirt project on-demand
+author: eedri
+tags: infrastructure, developers, ci
+date: 2016-12-20 10:00:00 CET
+---
+
+All projects in oVirt CI are built today post merge, using the 'build-artifacs' stage from [oVirt's CI standards](http://ovirt-infra-docs.readthedocs.io/en/latest/CI/Build_and_test_standards.html).
+
```
Comment (ifireball) [inline source/blog/2016-12-20-ci-please-build.html.md]: probably don't need this blank line here

File: source/blog/2016-12-20-ci-please-build.html.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+This ensures that all oVirt projects are built and deployed to oVirt repositories and can be consumed by CI jobs, developers or oVirt users.
```
Comment (ifireball) [inline source/blog/2016-12-20-ci-please-build.html.md]: Probably need a blank line here

File: source/blog/2016-12-20-ci-please-build.html.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+However, on some occasions a developer might need to build his project from an open patch in order to test the code before it is merged to the branch,
+So in any case of errors, he can go back to the patch and fix them before merging.
+Another exmaple is a UI developer, that works on a new UI fix/feature and wants to see the changes on each patch change.
```
Comment (ifireball) [inline source/blog/2016-12-20-ci-please-build.html.md]: Maybe:
However, on some occasions a developer might need to build his project from an open patch. Developers need this to be able to examine the effects of their changes on the full oVirt before merging those changes. On some cases developers may even want to hand over packages based on un-merged patches to the QE team to verify that a given change will fix some complex issue.

(The UI example is just the same thing - testing before merging)

File: source/blog/2016-12-20-ci-please-build.html.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+## The Current Build Option
+
+Until now, to build rpms from a patch, a developer needed to use a custom [Jenkins job](http://jenkins.ovirt.org/job/ovirt-engine_master_build-artifacts-el7-x86_64_build_from_patch/), which was only available to ovirt-engine and only for master branch. 
+Another option was to try and build it locally, but then you need to make sure you have all the right packages installed ( something that is given automatically in CI ).
```
Comment (ifireball) [inline source/blog/2016-12-20-ci-please-build.html.md]: A third option is to use mock_runner.sh (link to std-CI docs), but not all developers have the knowledge and resources to do that.

File: source/blog/2016-12-20-ci-please-build.html.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+It allows any oVirt developer to trigger a build from a patch just by adding the comment 'ci please build' on the patch in Gerrit. 
+It will then trigger a 'build-artifacts-on-demand' job on jenkins which will build rpms from the patch, but not publish them to any repo.
+
+## How Do I Enable It For My oVirt Project ?
+
+Adding this is a simple one liner to your project standard yaml file, just add the following line to your 'build-artifacts' section, under 'jobs':
+
+      - '{project}_{version}_{stage}-on-demand-{distro}-{arch}'
+
```
Comment (ifireball) [inline source/blog/2016-12-20-ci-please-build.html.md]: Maybe give example of path doing that here

File: source/blog/2016-12-20-ci-please-build.html.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+## What Else Can I Do With It ? (Hint: OST)
+
+Now that you know you can build new rpms whenever you want using a simple comment, you might ask yourself what can I do with those rpms?
```
Comment (ifireball) [inline source/blog/2016-12-20-ci-please-build.html.md]: One thing you can do, is run the oVirt system tests suit, to do that....

File: source/blog/2016-12-20-ci-please-build.html.md (hunk 4 above)
Comment (eedri) [inline source/blog/2016-12-20-ci-please-build.html.md]: Do you think its worth adding it to the blog? my guess most developer don't use it anyhow

**Pull request merged**